### PR TITLE
feat(OpenAPI): RHICOMPL-2077 rswag generator

### PR DESCRIPTION
**Pull request by romanblanco**
`podman-compose exec rails bundle exec rake rswag:specs:swaggerize`

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: spec/integration/profiles_spec.rb [hunk 1]
```diff
@@ -135,6 +161,15 @@
...
 
       response '201', 'creates a profile' do
+        before do
+          @parent.update(policy_id: @policy.id)
+          rule1 = FactoryBot.create(:rule, benchmark: @parent.benchmark, description: 'Benchmark rule 1')
+          rule2 = FactoryBot.create(:rule, benchmark: @parent.benchmark, description: 'Benchmark rule 2')
+          a1 = ProfileRule.new(rule: rule1, profile: @parent)
+          a1.save!
+          a2 = ProfileRule.new(rule: rule2, profile: @parent)
+          a2.save!
```
Comment (skateman) [inline spec/integration/profiles_spec.rb:171]:
```suggestion
          @parent.rules << rule1
          @parent.rules << rule2
```

File: lib/tasks/swaggerize.rake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      t.pattern = ENV.fetch(
+        'PATTERN',
+        'spec/integration/**/*_spec.rb'
+      )
+      t.rspec_opts = ['--format Rswag::Specs::SwaggerFormatter', '--order defined']
+    end
+  end
+end
+
+task rswag: ['rswag:specs:swaggerize']
```
Comment (skateman) [inline lib/tasks/swaggerize.rake:18]: Is this documented somewhere in our README?

**Review by skateman — CHANGES_REQUESTED** [review]

File: lib/tasks/swaggerize.rake (hunk 2 above)
Comment (romanblanco) [reply lib/tasks/swaggerize.rake:18]: It is: https://github.com/RedHatInsights/compliance-backend/blame/5338b2402ac5cf645dcdfb6d211fa47f00f20f4d/README.md#L357

**Comment by skateman** [comment]
@Victoremepunto wdyt? :grin: also @vkrizan? it has my :green_heart:

**Comment by skateman** [comment]
You can safely ignore the coverage issues...

**Comment by Victoremepunto** [comment]
> You can safely ignore the coverage issues...

How's that possible ? is it considering the spec file as code ? it should be ignored if that's the case

**Comment by skateman** [comment]
> > You can safely ignore the coverage issues...
> 
> How's that possible ? is it considering the spec file as code ? it should be ignored if that's the case

Check the report, it's the rake task ... we're not going to test the task that we copy-pasted from their GH repo.

**Review by vkrizan — APPROVED** [review]
LGTM

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**